code/datums/components/food/edible.dm

```diff
@@ -142,6 +159,95 @@ Behavior that's still missing from this component that original food items had t
 	qdel(our_atom)
 	return COMSIG_FRYING_HANDLED
 
+///Called when food is created through processing (Usually this means it was sliced). We use this to pass the OG items reagents.
+/datum/component/edible/proc/OnProcessed(datum/source, atom/original_atom, list/chosen_processing_option)
+	SIGNAL_HANDLER
+
+	if(!original_atom.reagents)
+		return
+
+	var/atom/this_food = parent
+	var/reagents_for_slice = chosen_processing_option[TOOL_PROCESSING_AMOUNT]
+
+	this_food.create_reagents(volume) //Make sure we have a reagent container
+
+	original_atom.reagents.trans_to(this_food, reagents_for_slice)
+
+	if(original_atom.name != initial(original_atom.name))
+		this_food.name = "slice of [original_atom.name]"
+	if(original_atom.desc != initial(original_atom.desc))
+		this_food.desc = "[original_atom.desc]"
+
+///Called when food is crafted through a crafting recipe datum.
+/datum/component/edible/proc/OnCraft(datum/source, list/parts_list, datum/crafting_recipe/food/recipe)
+	SIGNAL_HANDLER
+
+	var/atom/this_food = parent
+
+	this_food.reagents.clear_reagents()
+
+	for(var/obj/item/crafted_part in this_food.contents)
+		crafted_part.reagents?.trans_to(this_food.reagents, crafted_part.reagents.maximum_volume, CRAFTED_FOOD_INGREDIENT_REAGENT_MODIFIER)
+
+	if(istype(recipe))
+		contents_loop:
```

Comment:
Is the `istype()` check here expected?
Also not fan of this goto instruction.

Reply:
will look into fixing this

Reply:
and yeah the check is expected, its looking for the crafted parts in the contents so that it can keep them in, im not sure its weird code like most food code

Reply:
Add the check at the beginning of the proc, followed by a `CRASH()` with an informative message, instead.

Are you able to remove the goto instruction here on your own, or want me to code a suggestion instead?
It just needs a control variable added, like I explained to you on discord.

Reply:
it shouldnt crash its just deleting any non-ingredient items in the food. I can kill the goto though.

Reply:
on which situations is this signal triggered here, without a `recipe` variable of the proper type?

Reply:
oh i didnt know u meant that istype nvm